custom_components/aqara_gateway/__init__.py

```diff
@@ -104,6 +98,7 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
 
     gateway = hass.data[DOMAIN][entry.entry_id]
     gateway.stop()
+    await gateway.async_disconnect()
```

Comment:
cleans up the leftover thread (started by `_mqttc.loop_start()`) on entry unload.